report/report-server/src/main/kotlin/com/testerum/report_server/service/ReportService.kt

```diff
@@ -0,0 +1,42 @@
+package com.testerum.report_server.service
+
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.testerum.report_generators.reports.report_model.template.ManagedReportsExecutionListener
+import com.testerum.report_generators.reports.utils.EXECUTION_LISTENERS_OBJECT_MAPPER
+import com.testerum.runner.events.model.ConfigurationEvent
+import com.testerum.runner.events.model.RunnerEvent
+import org.springframework.stereotype.Service
+import java.nio.file.Path
+import java.nio.file.Paths
+
+@Service
+class ReportService() {
+
+    fun addReport(eventsList: List<String>) {
+        val configEvent: ConfigurationEvent = getConfigEvent(eventsList)
+
+        val reportPath = getReportPath(configEvent)
+
+        val managedReportsExecutionListener = ManagedReportsExecutionListener(reportPath)
+
+        managedReportsExecutionListener.start();
+        for (eventAsString in eventsList) {
+            val event = EXECUTION_LISTENERS_OBJECT_MAPPER.readValue<RunnerEvent>(eventAsString)
+            managedReportsExecutionListener.onEvent(event)
+        }
+        managedReportsExecutionListener.stop()
```

Comment:
We should call ``stop()`` in a finally block, like this:
```suggestion
        managedReportsExecutionListener.start();
        try {
            for (eventAsString in eventsList) {
                val event = EXECUTION_LISTENERS_OBJECT_MAPPER.readValue<RunnerEvent>(eventAsString)
                managedReportsExecutionListener.onEvent(event)
            }
        } finally {
            managedReportsExecutionListener.stop()
        }
```